### DISCUSSION TITLE
Zim "sections" in notebooks

### DISCUSSION THREAD

**Discussion by itsnotsaved** [zim-desktop-wiki/zim-desktop-wiki#1817] [Ideas]
Currently each notebooks open in separate instance but when open many notebooks then it's quite difficult to move between them within dock or taskbar.

Add option to open and move between multiple notebooks within same instance.

When open new notebook then provide option to choose for open within the instance or separate instance

**Comment by jaap-karssenberg** [comment]
What version are you looking at?  Latest versions by default open all
notebooks in a single process.

**Comment by itsnotsaved** [comment]
It seems i have asked the question incorrectly. Zim opens all notebooks in single process.
I concerned about each notebooks open in separate window (it also suitable; i have misunderstood once again)

I mean about onenote's "Group" feature, i'm not sure if there any existing issues for it
https://support.microsoft.com/en-us/office/create-a-section-group-in-onenote-47c33260-fdb8-4ae3-9c6b-39407bf1cba3

Each group can have each index (side panel) then if there are related contents/projects etc that difficult to maintain within single index so we can keep them all as groups without create completely separate notebooks.

The problem, i have so many main pages list with subpages; and i can't keep them in single notebook even if they're related to same project. Then i created separate notebooks although only related to one project while i have other projects notebooks as well then finally i have lot of opened notebook windows so i can't even recognize easily or find necessary notebook to maximize. 

I see onenote address this problem by offering "group" feature then it may be possible to implement for zim software.

Zim 0.73.5-t202105070815

**Comment by jaap-karssenberg** [comment]
Maybe I fail to understand the use case, but why not just add another level
of nesting the pages in your main notebook?  You can create as many levels
of sub-pages as you need to be organized.

-- Jaap

**Comment by itsnotsaved** [comment]
>You can create as many levels of sub-pages as you need to be organized.

yes i know it^^

I use "building" as example.

e.g. **[A Building]** let's say i have 40+ main pages with multiple subpages (e.g.5-10) for each. when has around 30 folders then it fills the left panel index screen in 15inch laptop screen (i know its possible to keep creating 60-70) but when has such amount in one index, it make complicate to switch to other pages or find them quick when working with multiple main pages (subpages in each etc) often. Furthermore when keep expanded several main pages then it makes super complicated

![Screenshot from 2021-06-19 03-40-04](https://user-images.githubusercontent.com/73447325/122653978-a6cf1c00-d172-11eb-9136-13cc9099569a.png)


The solution is create new notebooks for "A building" contents as 1st floor, 2nd floor, 3rd floor, 4th floor, 5th floor (so now already 6 notebooks for only "A building".
While need to keep notes same as for other 4 buildings and update them daily or nearly at same times. If that only 4 notebooks for 4 building then it's okey to keep them opened but when it 4x6=24 notebooks; you know it's hard to find and switch between 24 notebooks windows

![Screenshot from 2021-06-19 03-40-35](https://user-images.githubusercontent.com/73447325/122654064-04636880-d173-11eb-82f4-02c3a75f3e48.png)

Exactly feel complicated when someone keep notes about multiple scientific researches contents, multiple building contents, multiple large websites, wikis contents etc.

So i believe that's why onenote has feature called "sections/groups". it helps to keep multiple indexes inside one notebook (as keep multiple notebooks inside one notebook). The main goal is to have multiple indexes (side panels) in single notebook to reduce the list contain on single index; then as mentioned above each index able to use for each floor but all "A building" notes contain on single notebook.

https://support.microsoft.com/en-us/office/create-a-section-group-in-onenote-47c33260-fdb8-4ae3-9c6b-39407bf1cba3
The mentioned "groups" has on onenote 2016 (or older versions may be) however i believe "Section" feature more than enough to accomplish this goal rather than to be extended as groups

![image](https://user-images.githubusercontent.com/73447325/122656740-befd6600-d187-11eb-9d6b-a369936dc8d3.png)


If you need to test this feature; i'm not sure if you have already or if that easy to download and install old office product versions etc.

so i thought to add third-party tool https://github.com/itsnotsaved/PackageTool 
Usage: go to folder through cmd and enter `OfficeRTool` --> download [D] --> select office professional plus 2016 (it download through official setup storage --> install as default [I] --> convert retail to volume [C] --> now you can use properly for 30 days or add key otherwise use its key management [K] that helps to keep active for 180days (and must be refresh using it again as i know)

If you consider or interest to implement, i'll open new issue for this feature because this issue already has long descriptions and haven't pointed about expected feature properly within first comment shortly.

**Comment by jaap-karssenberg** [comment]
I can see that the section tabs as implemented in OneNote are visually
pleasing, I don't see how they would help you organize a large structure of
pages. The number of tabs you can see at any time is also limited, so I
don't see a clear benefit over having more levels of organization in the
index.

Maybe instead of describing the solution, please explain the problem first.
I understand you don't want a long list of main pages, but why not cut it
up in smaller lists by using another level of organization ?

Kind Regards,

Jaap

**Comment by itsnotsaved** [comment]
>Maybe instead of describing the solution, please explain the problem first.

Above i explained the problem but with other details, it may be not clear much.

>I understand you don't want a long list of main pages,

Yes

> but why not cut it, up in smaller lists by using another level of organization ?

I dont understand, do you mean multi-level subpages for each main page? I already have as it but when few main pages expanded then it make more longer list than when everything collapsed.

>please explain the problem first.

When has long list of main pages (with subpages) then i can't switch or find easily necessary pages when want to add details or read-back (then has collapse many pages or scroll down more or expand and look again, again). (see screenshot 1)
For solve it if i create new notebooks then next result is many opened notebooks windows on taskbar/dock; normally i can view only around 3-4 windows on average screen (see screenshot 2) then if has many opened notebooks, again it's not easy to find required notebook to switch.

>I can see that the section tabs as implemented in OneNote are visually pleasing, I don't see how they would help you organize a large structure of pages. The number of tabs you can see at any time is also limited

I don't think it is that much limited when consider about "one notebook"
(see screenshot 3) i have added numbers (1) (2) (3) (4) then atleast i can avoid to create another 3 notebooks so i see it as more beneficial and able to switch to another tab very quickly to see another category of pages list

Number of tabs see as limited because its title with tab interface "new section 1" in that screenshot but see this image

![image](https://user-images.githubusercontent.com/73447325/123249841-97095c00-d513-11eb-99e2-b7ded2d2b4bd.png)

It already display 10 tabs and still has more space.
If you consider to implement this feature; i think it's possible to move existing opened pages tabs to footer bar (it doesn't use for anything as i see; only empty) and introduce "sections" tabs to that top bar. (or you can display sections tabs in footer bar instead). I guess that will make this software more useful than today.

**Comment by jaap-karssenberg** [comment]
If I understand you correctly, the key problem is that you loose overview
in a multi-level index where everything is expanded. Would that be a good
summary?

Do you use the "automatic collapse" option for the pageindex plugin? That
might at least help a bit by collapsing again sections you are not using.

**Comment by itsnotsaved** [comment]
>the key problem is that you loose overview in a multi-level index where everything is expanded. Would that be a good summary?

Right and other-hand waste my time
(Even everything collapsed Including all subpages; i have to scroll down some to see all main pages) but i dont want make new notebooks for reduce and categorize them, as i mentioned above then become more difficult to handle when has separate notebooks windows and switch through taskbar/dock) then that's why i mentioned "sections" solve that problem perfectly. 

>Do you use the "automatic collapse" option for the pageindex plugin? That might at least help a bit by collapsing again sections you are not using.

Already in-use but it doesn't help to much reduce difficulties that face when maintain big notebooks

**Comment by introt** [comment]
Hi, joining in for some brainstorming, if you don't mind.

If I'm getting this correctly, you'd want separate tree views (ie. what the index is) for the sections the same way OneNote has in this screenshot:

![](https://user-images.githubusercontent.com/73447325/122656740-befd6600-d187-11eb-9d6b-a369936dc8d3.png)

So instead of the treeview showing the index for the whole notebook, it'd show the index for the subpages of the current section.

This could be implemented by having the Path Bar list the toplevel pages (eg. `:Section 1`, `:Section 2`, `:Quick notes`) and either making the Index plugin show the tree view starting from the toplevel page's subpages instead of `:`/`<toplevel>`, or implementing the alternative "partial tree view" plugin I've described [here](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1171#issuecomment-896922356).

So instead of the index looking like this as it currently does:

![image](https://user-images.githubusercontent.com/26069787/135152511-0a4ac843-355e-418f-ac57-f59515935a09.png)

It would look something like this, while retaining the same file structure:

![image](https://user-images.githubusercontent.com/26069787/135153522-77e245a7-b644-4c5e-a82b-2a8b7522d48f.png)
_Please ignore the wrong section being selected in the Path Bar_

Would this suit your needs?

**Comment by itsnotsaved** [comment]
I dont know if have misunderstood something (and from my screenshots) or not. and your comment seems bit complex so i add a small note instead ask questions, I expect to have onenote's section feature to put multiple notebooks in one zim notebook. then each section is a notebook with a index as showed on screenshot. i guess you've got it.

Anyway such features will convert zim to a feature-rich and powerful cross-platform opensource solution

![image](https://user-images.githubusercontent.com/73447325/147084909-810970e1-1faf-4557-849f-e4c18428d083.png)

**Comment by jaap-karssenberg** [comment]
This can easily be accomplished via a plugin. Both the pathbar an the index
can be re-used of existing plugins (imported, not copied). The index view
allows setting a toplevel in the API. The pathbar would need sub-class to
populate with the toplevels from the notebook. So probably plugin with very
little code needed to make this.

-- Jaap
